tools/auth/vk.go

```diff
@@ -0,0 +1,82 @@
+package auth
+
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/vk"
+)
+
+var _ Provider = (*Vk)(nil)
+
+// NameVk is the unique name of the Vk provider.
+const NameVk string = "vk"
+
+// Vk allows authentication via Vk OAuth2.
+type Vk struct {
+	*baseProvider
+}
+
+// NewVkProvider creates new Vk provider instance with some defaults.
+//
+// Docs: https://dev.vk.com/api/oauth-parameters
+func NewVkProvider() *Vk {
+	return &Vk{&baseProvider{
+		scopes:     []string{"email"},
+		authUrl:    vk.Endpoint.AuthURL,
+		tokenUrl:   vk.Endpoint.TokenURL,
+		userApiUrl: "https://api.vk.com/method/users.get?fields=photo_max,screen_name&v=5.131",
+	}}
+}
+
+// FetchAuthUser returns an AuthUser instance based on Vk's user api.
+//
+// API reference: https://dev.vk.com/method/users.get
+func (p *Vk) FetchAuthUser(token *oauth2.Token) (*AuthUser, error) {
+	data, err := p.FetchRawUserData(token)
+	if err != nil {
+		return nil, err
+	}
+
+	rawUser := map[string]any{}
+	if err := json.Unmarshal(data, &rawUser); err != nil {
+		return nil, err
+	}
+
+	extracted := struct {
+		Response []struct {
+			Id        int    `json:"id"`
+			FirstName string `json:"first_name"`
+			LastName  string `json:"last_name"`
+			Username  string `json:"screen_name"`
+			AvatarUrl string `json:"photo_max"`
```

Comment:
I think `deactivated` field can only be returned if requested user is not current one. If you request data about current user and it's deactivated it should return error at `FetchRawUserData` cause it would be unauthorized. I think it's error [18 - Page is deleted or banned](https://dev.vk.com/reference/errors).